processing/src/main/java/io/druid/segment/DimensionHandler.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.metamx.common.io.smoosh.FileSmoosher;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.query.groupby.GroupByQueryEngine;
+import io.druid.segment.column.Column;
+import io.druid.segment.column.ColumnCapabilities;
+import io.druid.segment.column.ValueType;
+import io.druid.segment.data.IOPeon;
+import io.druid.segment.data.IndexedInts;
+
+import java.io.Closeable;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Processing related interface
+ *
+ * A DimensionHandler is an object that encapsulates indexing, column merging/building, and querying operations
+ * for a given dimension type (e.g., dict-encoded String, Long).
+ *
+ * These operations are handled by sub-objects created through a DimensionHandler's methods:
+ *   DimensionIndexer, DimensionMerger, and DimensionColumnReader, respectively.
+ *
+ * Each DimensionHandler object is associated with a single dimension.
+ *
+ * This interface allows type-specific behavior column logic, such as choice of indexing structures and disk formats.
+ * to be contained within a type-specific set of handler objects, simplifying processing classes
+ * such as IncrementalIndex and IndexMerger and allowing for abstracted development of additional dimension types.
+ *
+ * A dimension may have two representations, an encoded representation and a actual representation.
+ * For example, a value for a String dimension has an integer dictionary encoding, and an actual String representation.
+ *
+ * A DimensionHandler is a stateless object, and thus thread-safe; its methods should be pure functions.
+ *
+ * @param <EncodedType> class of the encoded values
+ * @param <ActualType> class of the actual values
+ */
+public interface DimensionHandler<EncodedType extends Comparable<EncodedType>, EncodedTypeArray, ActualType extends Comparable<ActualType>>
```

Comment:
Mostly I'm curious because you can either have items which are by their nature Comparable, or have some other item in the chain take a Comparator as an argument. And it is not clear that one is preferable over the other.